azurerm/internal/services/keyvault/key_vault_certificate_issuer_resource.go

```diff
@@ -0,0 +1,329 @@
+			if err != nil {
+				return fmt.Errorf("failed to flatten Azure KeyVault Certificate Issuer Admin Details: %v", err)
+			}
+			d.Set("admin", adminDetails)
+		}
+	}
+	if resp.Credentials != nil {
+		if resp.Credentials.AccountID != nil {
+			d.Set("account_id", resp.Credentials.AccountID)
+		}
+		if resp.Credentials.Password != nil {
+			d.Set("password", resp.Credentials.Password)
+		}
+	}
+
+	return nil
+}
+
+func resourceArmKeyVaultCertificateIssuerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).KeyVault.ManagementClient
+	keyVaultClient := meta.(*clients.Client).KeyVault.VaultsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.KeyVaultCertificateIssuerId(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// we verify it exists
+	keyVaultId, err := azure.GetKeyVaultIDFromBaseUrl(ctx, keyVaultClient, id.KeyVaultBaseUrl)
+	if err != nil {
+		return fmt.Errorf("Error retrieving the Resource ID the Key Vault at URL %q: %s", id.KeyVaultBaseUrl, err)
+	}
+
+	ok, err := azure.KeyVaultExists(ctx, keyVaultClient, *keyVaultId)
+	if err != nil {
+		return fmt.Errorf("Error checking if key vault %q for Issuer %q in Vault at url %q exists: %v", *keyVaultId, id.Name, id.KeyVaultBaseUrl, err)
+	}
+
+	if !ok {
+		log.Printf("[DEBUG] Issuer %q (Key Vault %q) was not found in Key Vault at URI %q - removing from state", id.Name, *keyVaultId, id.KeyVaultBaseUrl)
+		d.SetId("")
+		return nil
+	}
+
+	_, err = client.DeleteCertificateIssuer(ctx, id.KeyVaultBaseUrl, id.Name)
+
+	return err
+}
+
+func expandKeyVaultCertificateIssuerOrganizationDetailsAdminDetails(vs []interface{}) *[]keyvault.AdministratorDetails {
+	results := make([]keyvault.AdministratorDetails, 0, len(vs))
+
+	for _, v := range vs {
+		administratorDetails := keyvault.AdministratorDetails{}
+		args := v.(map[string]interface{})
+		if firstName, ok := args["first_name"]; ok {
+			administratorDetails.FirstName = utils.String(firstName.(string))
+		}
+		if lastName, ok := args["last_name"]; ok {
+			administratorDetails.LastName = utils.String(lastName.(string))
+		}
+		if emailAddress, ok := args["email_address"]; ok {
+			administratorDetails.EmailAddress = utils.String(emailAddress.(string))
+		}
+		if phone, ok := args["phone"]; ok {
+			administratorDetails.Phone = utils.String(phone.(string))
+		}
+		results = append(results, administratorDetails)
+	}
+
+	return &results
+}
+
+func flattenKeyVaultCertificateIssuerAdmins(input *[]keyvault.AdministratorDetails) ([]interface{}, error) {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results, nil
+	}
+
+	for k, admin := range *input {
+		result := make(map[string]interface{})
+		if admin.EmailAddress != nil {
+			result["email_address"] = admin.EmailAddress
+		} else {
+			return nil, fmt.Errorf("required attribute `email_address` missing in API response for admin index %d", k)
+		}
```

Comment:
this isn't a user actionable error - so can we remove this?

Reply:
We can either continue to gather all valid values, or we can return an empty slice, or we can panic. Which do you prefer?

Reply:
I've opted to just continue on.